pkg/controller/servingcert/controller/secret_creating_controller.go

```diff
@@ -134,63 +134,27 @@ func (sc *serviceServingCertController) generateCert(serviceCopy *v1.Service) er
-		_, updateErr := sc.serviceClient.Services(serviceCopy.Namespace).Update(serviceCopy)
-
-		// if we're past the max retries and we successfully updated, then the sync loop successfully handled this service and we want to forget it
-		if updateErr == nil && getNumFailures(serviceCopy) >= sc.maxRetries {
-			return nil
-		}
-		return err
+		return sc.updateServiceFailure(serviceCopy, err)
 	}
 	if kapierrors.IsAlreadyExists(err) {
 		actualSecret, err := sc.secretClient.Secrets(serviceCopy.Namespace).Get(secret.Name, metav1.GetOptions{})
 		if err != nil {
-			// if we have an error creating the secret, then try to update the service with that information.  If it fails,
-			// then we'll just try again later on  re-list or because the service had already been updated and we'll get triggered again.
-			serviceCopy.Annotations[api.ServingCertErrorAnnotation] = err.Error()
-			serviceCopy.Annotations[api.AlphaServingCertErrorAnnotation] = err.Error()
-			numFailure := strconv.Itoa(getNumFailures(serviceCopy) + 1)
-			serviceCopy.Annotations[api.AlphaServingCertErrorNumAnnotation] = numFailure
-			serviceCopy.Annotations[api.ServingCertErrorNumAnnotation] = numFailure
-			_, updateErr := sc.serviceClient.Services(serviceCopy.Namespace).Update(serviceCopy)
-
-			// if we're past the max retries and we successfully updated, then the sync loop successfully handled this service and we want to forget it
-			if updateErr == nil && getNumFailures(serviceCopy) >= sc.maxRetries {
-				return nil
-			}
-			return err
+			return sc.updateServiceFailure(serviceCopy, err)
 		}
 
-		if (actualSecret.Annotations[api.AlphaServiceUIDAnnotation] != string(serviceCopy.UID)) && (actualSecret.Annotations[api.ServiceUIDAnnotation] != string(serviceCopy.UID)) {
-			serviceCopy.Annotations[api.AlphaServingCertErrorAnnotation] = fmt.Sprintf("secret/%v references serviceUID %v, which does not match %v", actualSecret.Name, actualSecret.Annotations[api.AlphaServiceUIDAnnotation], serviceCopy.UID)
-			serviceCopy.Annotations[api.ServingCertErrorAnnotation] = fmt.Sprintf("secret/%v references serviceUID %v, which does not match %v", actualSecret.Name, actualSecret.Annotations[api.ServiceUIDAnnotation], serviceCopy.UID)
-			numFailure := strconv.Itoa(getNumFailures(serviceCopy) + 1)
-			serviceCopy.Annotations[api.ServingCertErrorNumAnnotation] = numFailure
-			serviceCopy.Annotations[api.AlphaServingCertErrorNumAnnotation] = numFailure
-			_, updateErr := sc.serviceClient.Services(serviceCopy.Namespace).Update(serviceCopy)
-
-			// if we're past the max retries and we successfully updated, then the sync loop successfully handled this service and we want to forget it
-			if updateErr == nil && getNumFailures(serviceCopy) >= sc.maxRetries {
-				return nil
-			}
-			// TODO: Return ServingCertErrorAnnotation when removing alpha annotations.
-			return errors.New(serviceCopy.Annotations[api.AlphaServingCertErrorAnnotation])
+		if notSameUID(actualSecret, serviceCopy) {
```

Comment:
Can you please make it just `uidsEqual()`? `!uidsEqual()` reads better than a possible future `!notSameUID()`.